api/buildings/models.py

```diff
@@ -248,3 +257,72 @@ class Meta:
 
     def __str__(self):
         return self.name
+
+
+class ApprovedImage(models.Manager):
+    def get_queryset(self):
+        return super().get_queryset().filter(status=ImageFile.ACCEPTED)
+
+
+class ImageFile(models.Model):
+    PENDING = 0
+    ACCEPTED = 1
+    REJECTED = -1
+
+    IMAGE_STATUS_CHOICES = [
+        (PENDING, _("Pending")),
+        (ACCEPTED, _("Accepted")),
+        (REJECTED, _("Rejected")),
+    ]
+
+    def set_images_limit(building):
+        if ImageFile.objects.filter(building_id=building).count() >= settings.ALLOWED_IMAGES_LIMIT:
+            raise ValidationError("Image limit for building is reached (%s)" % settings.ALLOWED_IMAGES_LIMIT)
+
+    def image_thumb(self):
+        return mark_safe(
+            '<a href={0}><img src="{0}" url width="50" height="50" /></a>'.format(
+                os.path.join(settings.MEDIA_URL, str(self.image))
```

Comment:
Why not `self.image.url` ?

Reply:
I was not aware of this attribute, this is now fixed.